File: app/src/jvx/java/art/coded/wireframe/viewmodel/CustomViewModel.java
package art.coded.wireframe.viewmodel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class CustomViewModel extends ViewModel {

    static final String LOG_TAG = CustomViewModel.class.getSimpleName();

    private final MutableLiveData<String> mText;

    public CustomViewModel() {
        mText = new MutableLiveData<>();
        mText.setValue("This is custom view");
    }

    public LiveData<String> getText() {
        return mText;
    }
}

File: app/src/ktx/java/art/coded/wireframe/viewmodel/CustomViewModel.kt
package art.coded.wireframe.viewmodel

import androidx.lifecycle.ViewModel
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModelProvider

private val LOG_TAG = CustomViewModel::class.java.simpleName

class CustomViewModel : ViewModel() {

    val text = MutableLiveData<String>()

    init {
        text.value = "This is a custom view"
    }
}

class CustomViewModelFactory() : ViewModelProvider.Factory {
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return CustomViewModel() as T
    }
}